File: Count-Equal-and-Divisible-Pairs-in-an-Array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countPairs = function (nums, k) {
    let count = 0;
    const n = nums.length;
    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            if (nums[i] === nums[j] && (i * j) % k === 0) {
                count++;
            }
        }
    }
    return count;
};

File: Count-Equal-and-Divisible-Pairs-in-an-Array.py
class Solution(object):
    def countPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        count = 0
        n = len(nums)
        for i in range(n):
            for j in range(i + 1,  n):
                if (nums[i] == nums[j] and (i * j)% k == 0):
                    count = count + 1
        
        return count 